node/src/components/deploy_buffer.rs

```diff
@@ -94,6 +102,57 @@ impl DeployBuffer {
         })
     }
 
+    pub(crate) fn initialize_component(
+        &mut self,
+        effect_builder: EffectBuilder<MainEvent>,
+        storage: &Storage,
+    ) -> Option<Effects<MainEvent>> {
+        if <Self as InitializedComponent<MainEvent>>::is_uninitialized(self) {
+            info!(
+                "pending initialization of {}",
+                <Self as Component<MainEvent>>::name(self)
+            );
+            <Self as InitializedComponent<MainEvent>>::set_state(
+                self,
+                ComponentState::Initializing,
+            );
+            let blocks = match storage.read_blocks_for_replay_protection() {
+                Ok(blocks) => blocks,
+                Err(err) => {
+                    return Some(
+                        fatal!(
+                            effect_builder,
+                            "fatal block store error when attempting to read highest blocks: {}",
+                            err
+                        )
+                        .ignore(),
+                    )
+                }
+            };
+            debug!(
+                blocks = ?blocks.iter().map(|b| b.height()).collect_vec(),
+                "DeployBuffer: initialization"
+            );
+            info!("initialized {}", <Self as Component<MainEvent>>::name(self));
+            let event = Event::Initialize(blocks);
+            return Some(smallvec![async {
+                smallvec![MainEvent::DeployBuffer(event)]
+            }
+            .boxed()]);
```

Comment:
Can't we do `wrap_effects` or something similar that is more consistent with the other `Event`/`Effects` code?

Reply:
This is actually consistent with other places when we initialize components, like [here](https://github.com/casper-network/casper-node/blob/e091dbfd1b0cce7d40e7e0b92935bdb0746acc79/node/src/reactor/main_reactor/utils.rs#L19) or [here](https://github.com/casper-network/casper-node/blob/e091dbfd1b0cce7d40e7e0b92935bdb0746acc79/node/src/components/network/tests.rs#L206).

I'm open to changing it, but not as a part of this PR - it has already grown much beyond initially planned scope.

Reply:
I meant consistent with the code base in general, but I see what you mean and I agree, let's not do it in this PR.